### ISSUE TITLE
Weighted Metadata Searches / Multiple Providers Used

### ISSUE THREAD

**Issue by mcrosson** [krateng/maloja#23]
Would it be possible to assign a 'weight' to the different metadata providers with a cascade through them?

I Just discovered some artists have cover art and other metadata on different providers and Maloja doesn't seem to move onto a different provider when one fails.

I'd also like to order the providers by my personal preference as I've found certain providers have more 'vanilla' cover art than others.

**Comment by krateng** [comment]
It should fallback onto other providers when one fails - does that not happen for you? Could you check the relevant entries in your `logs/thirdparty.log` file?

The sorting is definitely a good idea, I'll give it a try.

**Comment by mcrosson** [comment]
The fallback didn't work on my deployment. I ended up disabling LastPass as a provider to get around the artist image that wasn't found. Looking closer at the logs it looks like a few tracks also are failing to get images and not flowing onto the next provider(s) either.

```

2020/07/28 16:07:03  Could not get track image for (['MASTER BOOT RECORD'], 'DMA 3 LPT1 ECP MODE') from LastFM
2020/07/28 16:26:06  Could not get artist image for Slipknot from LastFM
2020/07/28 16:26:06  Got track image for (['Slipknot'], 'Psychosocial') from LastFM
2020/07/28 16:26:06  Got track image for (['Slipknot'], 'Killpop') from LastFM
2020/07/28 16:26:07  Got track image for (['Slipknot'], 'The Shape') from LastFM
2020/07/28 16:26:24  Could not get artist image for Slipknot from LastFM
2020/07/28 16:26:25  Could not get artist image for Slipknot from LastFM
2020/07/28 16:26:34  Could not get artist image for Slipknot from LastFM
2020/07/28 16:27:36  Could not get artist image for Slipknot from LastFM
2020/07/28 16:27:37  Got track image for (['Slipknot'], 'This Cold Black') from LastFM
2020/07/28 16:32:12  Spotify registered as metadata provider
2020/07/28 16:32:12  MusicBrainz registered as metadata provider
2020/07/28 16:32:49  Got artist image for Slipknot from Spotify

```

**Comment by krateng** [comment]
Custom sorting should now be possible, maybe that fixes the issue anyway. Let me know if it works!

**Comment by mcrosson** [comment]
I'm getting a runtime error. I jus trebuilt my docker container and it looks like 2.6.2 was installed correctly inside the container.

Error log:
```
Traceback (most recent call last):
  File "/usr/local/bin/maloja", line 5, in <module>
    from maloja.proccontrol.control import main
  File "/usr/local/lib/python3.8/site-packages/maloja/proccontrol/control.py", line 9, in <module>
    from . import tasks
ImportError: cannot import name 'tasks' from 'maloja.proccontrol' (/usr/local/lib/python3.8/site-packages/maloja/proccontrol/__init__.py)
```

**Comment by krateng** [comment]
Sorry for that, had some issues with packaging - 2.6.4 work.

**Comment by mcrosson** [comment]
Thanks for the update!

Looks like it's working now. It's honoring the priority order I specified and is skipping Last.FM, exactly how I have it configured.

👍 

```

2020/07/28 19:39:47  LastFM registered as metadata provider
2020/07/28 19:39:47  Spotify registered as metadata provider
2020/07/28 19:39:47  MusicBrainz registered as metadata provider
2020/07/28 19:40:33  Could not get track image for (['Slipknot'], 'Everything Ends') from MusicBrainz
2020/07/28 19:40:33  Got track image for (['Slipknot'], 'Everything Ends') from Spotify
2020/07/28 19:40:35  Could not get track image for (['Slipknot'], 'Dead Memories') from MusicBrainz
2020/07/28 19:40:35  Got track image for (['Slipknot'], 'Dead Memories') from Spotify
2020/07/28 19:40:38  Got track image for (['Slipknot'], 'AOV') from MusicBrainz
2020/07/28 19:40:42  Got track image for (['Slipknot'], 'Before I Forget') from MusicBrainz
2020/07/28 19:40:44  Could not get track image for (['Slipknot'], 'Override') from MusicBrainz
2020/07/28 19:40:45  Got track image for (['Slipknot'], 'Override') from Spotify

```

**Comment by mcrosson** [comment]
Closing as this issue has been addressed in 2.6.4
